bad_cert_demo_jc.js

```diff
@@ -0,0 +1,24 @@
+var tls = require('tls');
+var https = require('https');
+
+var socket1 = tls.connect({
+    port: 1337,
+    host: 'https://example1.com',
+    rejectUnauthorized: true,
+    checkServerIdentity: function(servername, peer) {  
```

Comment:
<!-- Comment managed by coverity-github-pr-commenter, do not modify!
4f3d9b5b5516c3a8705f4ccbf3a4d5da
-->
# Coverity Issue - TLS uses bad certificate verification
_Medium CWE-295_
The "checkServerIdentity" property in the "tls.connect()" function uses bad cert verification.

Improper certificate verification may allow an attacker to spoof a trusted entity by using a man-in-the-middle attack.
## How to fix
Use the default implementation either implicitly or explicitly, as the default function provided by the "tls" module is secure. When using a custom implementation, make sure that the function throws an error when the certificate does not match, and not just returns "undefined" on every input.